src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1229,22 +1229,24 @@
         <summary>Gets the "async" token.</summary>
       </PropertyComment>
     </Field>
-    <Field Name="Block" Type="BlockSyntax" Optional="true">
-      <PropertyComment>
-        <summary>
-          BlockSyntax node representing the body of the anonymous function.
-          Only one of Block or ExpressionBody will be non-null.
-        </summary>
-      </PropertyComment>
-    </Field>
-    <Field Name="ExpressionBody" Type="ExpressionSyntax" Optional="true">
-      <PropertyComment>
-        <summary>
-          ExpressionSyntax node representing the body of the anonymous function.
-          Only one of Block or ExpressionBody will be non-null.
-        </summary>
-      </PropertyComment>
-    </Field>
+    <Choice>
+      <Field Name="Block" Type="BlockSyntax">
+        <PropertyComment>
+          <summary>
+            BlockSyntax node representing the body of the anonymous function.
+            Only one of Block or ExpressionBody will be non-null.
+          </summary>
+        </PropertyComment>
+      </Field>
+      <Field Name="ExpressionBody" Type="ExpressionSyntax">
```

Comment:
note: `Optional=true` is removed here because it's redundant with `<Choice>`.  It is added back into the data model later on when we flatten these nodes into just a list of fields.  This makes things simpler for downstream tools, as they can just generate `(Block | Expression)` as opposed to `(Block? | Expression?)`.  The former properly indicates we want one of these.  THe latter indicates that not having both is ok.  The former is strictly more accurate and produces better results.

Thanks!